File: java-backend/src/main/java/com/digitalheroes/model/Product.java
package com.digitalheroes.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import java.util.Date;

@Data
@Document(collection = "products")
public class Product {
    @Id
    private String id;
    private String name;
    private double price;
    private double originalPrice;
    private String imageUrl;
    private String category;
    private double rating;
    private int reviews;
    private Date createdAt = new Date();
}

File: models/Product.js
const mongoose = require("mongoose");

const productSchema = new mongoose.Schema({
  name: { type: String, required: true },
  category: { type: String, required: true },
  price: { type: Number, required: true },
  originalPrice: { type: Number },
  imageUrl: { type: String, required: true },
  rating: { type: Number, default: 4.5 },
  reviews: { type: Number, default: 120 },
  vendorName: { type: String, default: "Digital Heroes Official" },
  isApproved: { type: Boolean, default: true }, // default true for internal, false for submissions
  createdAt: { type: Date, default: Date.now },
});

module.exports = mongoose.model("Product", productSchema);
